pkg/codegen/schema/schema.go

```diff
@@ -284,7 +284,200 @@ type Package struct {
 	// Functions is the list of functions defined by the package.
 	Functions []*Function
 	// Language specifies additional language-specific data about the package.
-	Language map[string]json.RawMessage
+	Language map[string]interface{}
+}
+
+// Language provides hooks for importing language-specific metadata in a package.
+type Language interface {
+	// ImportDefaultSpec decodes language-specific metadata associated with a Default.
+	ImportDefaultSpec(def *DefaultValue, bytes json.RawMessage) (interface{}, error)
+	// ImportPropertySpec decodes language-specific metadata associated with a Property.
+	ImportPropertySpec(property *Property, bytes json.RawMessage) (interface{}, error)
+	// ImportObjectTypeSpec decodes language-specific metadata associated with a ObjectType.
+	ImportObjectTypeSpec(object *ObjectType, bytes json.RawMessage) (interface{}, error)
+	// ImportResourceSpec decodes language-specific metadata associated with a Resource.
+	ImportResourceSpec(resource *Resource, bytes json.RawMessage) (interface{}, error)
+	// ImportFunctionSpec decodes language-specific metadata associated with a Function.
+	ImportFunctionSpec(function *Function, bytes json.RawMessage) (interface{}, error)
+	// ImportPackageSpec decodes language-specific metadata associated with a Package.
+	ImportPackageSpec(pkg *Package, bytes json.RawMessage) (interface{}, error)
+}
+
+func sortedLanguageNames(metadata map[string]interface{}) []string {
+	names := make([]string, 0, len(metadata))
+	for lang := range metadata {
+		names = append(names, lang)
+	}
+	sort.Strings(names)
+	return names
+}
+
+func importDefaultLanguages(def *DefaultValue, languages map[string]Language) error {
+	for _, name := range sortedLanguageNames(def.Language) {
+		val := def.Language[name]
+		if raw, ok := val.(json.RawMessage); ok {
+			if lang, ok := languages[name]; ok {
+				val, err := lang.ImportDefaultSpec(def, raw)
+				if err != nil {
+					return errors.Wrapf(err, "importing %v metadata", name)
+				}
+				def.Language[name] = val
+			}
+		}
+	}
+	return nil
+}
+
+func importPropertyLanguages(property *Property, languages map[string]Language) error {
+	if property.DefaultValue != nil {
+		if err := importDefaultLanguages(property.DefaultValue, languages); err != nil {
+			return errors.Wrapf(err, "importing default value")
+		}
+	}
+
```

Comment:
Should we also handle `ConstValue` here?

Reply:
Ah, yes.

Reply:
Oh, wait--we don't have language-specific info for consts. We'll need to if/when we do.